File: apps/api/AssetManagement.Api/DTOs/SettingsDtos.cs
namespace AssetManagement.Api.DTOs;

public record SystemSettingsDto(
    string OrgName,
    string Currency,
    string DateFormat,
    int DefaultPageSize
);

public record AlertSettingsDto(
    bool WarrantyEnabled,
    bool CertificateEnabled,
    bool LicenceEnabled,
    string Thresholds,
    string SmtpHost,
    int SmtpPort,
    string SmtpUsername,
    string SmtpPassword,
    string SmtpFromAddress,
    string SlackWebhookUrl,
    string Recipients
);

File: apps/api-kt/src/main/kotlin/com/assetmanagement/api/dto/SettingsDtos.kt
package com.assetmanagement.api.dto

data class SystemSettingsDto(
    val orgName: String,
    val currency: String,
    val dateFormat: String,
    val defaultPageSize: Int
)

data class AlertSettingsDto(
    val warrantyEnabled: Boolean,
    val certificateEnabled: Boolean,
    val licenceEnabled: Boolean,
    val thresholds: String,
    val emailProvider: String = "smtp",
    val smtpHost: String,
    val smtpPort: Int,
    val smtpUsername: String,
    val smtpPassword: String,
    val smtpFromAddress: String,
    val graphTenantId: String = "",
    val graphClientId: String = "",
    val graphClientSecret: String = "",
    val graphFromAddress: String = "",
    val slackWebhookUrl: String,
    val slackWarrantyWebhookUrl: String = "",
    val slackCertificateWebhookUrl: String = "",
    val slackLicenceWebhookUrl: String = "",
    val recipients: String,
    val scheduleType: String = "disabled",
    val scheduleTime: String = "09:00",
    val scheduleDay: String = "MONDAY"
)
